frontend/src/metabase/dashboard/reducers.js

```diff
@@ -372,14 +371,14 @@ const loadingDashCards = handleActions(
       }),
     },
     [FETCH_DASHBOARD_CARD_DATA]: {
-      next: (state, { payload: { currentTime, dashcardIds } }) => {
-        const loadingIds = Array.isArray(dashcardIds) ? dashcardIds : [];
-
+      next: (state, { payload: { currentTime, loadingIds } }) => {
+        const newLoadingIds = state.loadingIds.concat(loadingIds);
         return {
           ...state,
-          dashcardIds: loadingIds,
-          loadingIds,
-          loadingStatus: loadingIds.length > 0 ? "running" : "idle",
+          // used for updating documents title. It's used for the number of total request loaded.
+          // dashcardIds: loadingIds,
+          loadingIds: newLoadingIds,
+          loadingStatus: newLoadingIds.length > 0 ? "running" : "idle",
           startTime: loadingIds.length > 0 ? currentTime : null,
```

Comment:
I still need to figure out how to deal with `startTime` this isn't done yet. `startTime` is used to determine if we want to flag a dashboard slow.